c/zss.c

```diff
@@ -698,37 +698,81 @@ static void installWebPluginDefintionsService(WebPluginListElt *webPlugins, Http
-  
-  Json *json = jsonParseFile(slh, serverConfigFile, errorBuffer, sizeof(errorBuffer));
-  if (json) {
-    if (jsonIsObject(json)) {
-      JsonObject *jsonObject = jsonAsObject(json);
-      JsonObject *logLevels = jsonObjectGetObject(jsonObject, "logLevels");
-      if (logLevels != NULL) {
-        JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
-        while (property != NULL) {
-          if (!strcmp(jsonPropertyGetKey(property), pluginIdentifier)) {
-            int logLevel = jsonObjectGetNumber(logLevels, pluginIdentifier);
-            if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
-              return logLevel;
-            }
-            else {
-              break;
-            }
-          }
-          property = jsonObjectGetNextProperty(property);
+#define ENV_LOGLEVELS_PREFIX "ZWED_logLevels_"
+static int getLogLevelsFromEnv(JsonObject *envConfig, const char *identifier) {
+  int len = strlen(ENV_LOGLEVELS_PREFIX) + strlen(identifier) + 1;
+  int logLevel = ZOWE_LOG_SEVERE;
+  char envKey[len];
+  memset(envKey,'\0',len);
+  snprintf(envKey, len, "%s%s", ENV_LOGLEVELS_PREFIX, identifier);
+  logLevel = jsonObjectGetNumber(envConfig, envKey);
+  return logLevel;
+}
+
+static int checkLoggingVerbosity(JsonObject *serverConfig, JsonObject *envConfig, char *pluginIdentifier) {
+  int logLevel = getLogLevelsFromEnv(envConfig, pluginIdentifier);
+  if (!logLevel) {
+    JsonObject *logLevels = jsonObjectGetObject(serverConfig, "logLevels");
+    if (logLevels != NULL) {
+      JsonProperty *property = jsonObjectGetFirstProperty(logLevels);
+      while (property != NULL) {
+        if (!strcmp(jsonPropertyGetKey(property), pluginIdentifier)) {
+          logLevel = jsonObjectGetNumber(logLevels, pluginIdentifier);
+          break;
         }
+        property = jsonObjectGetNextProperty(property);
       }
     }
   }
-  
+
+  if (logLevel <= ZOWE_LOG_DEBUG3 || logLevel >= ZOWE_LOG_ALWAYS) {
+    return logLevel;
+  }  
   return ZOWE_LOG_INFO;
 }
 
+static char *formatDataServiceIdentifier(const char *unformattedIdentifier, size_t identifierLength, char idNameSeparator)
+{
+  char *dataServiceIdentifier = safeMalloc(identifierLength, "dataServiceIdentifier");
+  if (dataServiceIdentifier) {
+    memset(dataServiceIdentifier, 0, identifierLength);
+    strncpy(dataServiceIdentifier, unformattedIdentifier, identifierLength);
+    for (int i = 0; dataServiceIdentifier[i]; i++) {
+      if (dataServiceIdentifier[i] == '/') {
+        dataServiceIdentifier[i] = idNameSeparator;
+      }
+    }
+  }
+  return dataServiceIdentifier;
+}
+
+static void checkAndSetDataServiceLoglevel(JsonObject * serverConfig,
+                                           JsonObject * envConfig,
+                                           char *dataServiceIdentifier, 
+                                           uint64 loggingIdentifier) {
+  int logLevel = getLogLevelsFromEnv(envConfig, dataServiceIdentifier);
```

Comment:
Same as above.

Reply:
Added jsonObjectHasKey() check in getLogLevelsFromEnv(). So if env is not set then getLogLevelsFromEnv() returns ZOWE_LOG_NA.